### DISCUSSION TITLE
Use external (installed directly on the server) MySQL server

### DISCUSSION THREAD

**Discussion by mihha** [photoprism/photoprism#1059] [Q&A] [answered]
Hi,

I would like to use Photoprism but would like to remove MariaDB from docker-compose file because I already have MySQL server installed and optimized on the host where I would like to run Photoprism in container

Can you guys suggest the changes in docker-compose to achieve what I am looking for?

Thank you in advance!

**Comment by lastzero** [comment]
Simply remove the `mariadb:` part at the bottom and change `PHOTOPRISM_DATABASE_SERVER` to whatever server you want to use. Our examples for reference:

https://dl.photoprism.org/docker/

**Reply by mihha** [reply]
I already tried that but it always reports that it fails to connect to the localhost

Is there some network configuration that I need to adjust in the docker compose so photoprism can "see" the network from host?

**Reply by lastzero** [reply]
PhotoPrism runs inside a container, so the database server can not be localhost - unless you're running photoprism without Docker. Does your database server have a hostname (or IP) other than "localhost", which can not be used from outside your host?

**Reply by mihha** [reply]
My DB server can be accessed only from localhost on the host machine

It is not accessible from the outside world using its IP address

**Comment by alexisLefebvre** [comment, marked as answer]
If your host server runs Linux, use `172.17.0.1` to access to the network of the host: https://docs.docker.com/network/network-tutorial-standalone/#use-user-defined-bridge-networks

**Comment by StephenBrown2** [comment]
Since docker `20.04` or `20.10`, you can use the following entry in your service definition to access the local host:
```
extra_hosts:
- "host.docker.internal:host-gateway"
```
Then set `PHOTOPRISM_DATABASE_SERVER: "host.docker.internal:3306"`

I literally just set this up ten minutes ago, though I had to use `172.17.0.1` instead of `host-gateway` as my docker is still at 19.03.
